schemas/stsci.edu/asdf/wcs/frame-0.1.0.yaml

```diff
@@ -0,0 +1,47 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/wcs/frame-0.1.0"
+title: >
+  The base class of all coordinate frames.
+description: >
+  These objects are designed to be nested in arbitrary ways to build up
+  transformation pipelines out of a number of low-level pieces.
+type: object
+properties:
+  name:
+    description: |
+      A user-friendly name for the frame.
+    type: string
+
+  axes_order:
+    description: |
+      The order of the axes.
+    type: array
+    items:
+      type: integer
+
+  axes_names:
+    description: |
+      The name of each axis in this frame.
+    type: array
+    items:
+      anyOf:
+        - type: string
+        - type: 'null'
+
+  reference_frame:
+    description: |
+      The reference frame.
+    enum: [ICRS, FK5, FK4, FK4_noeterms, galactic, galactocentric,
+           altaz, GCRS, CIRS, ITRS, precessed_geocentric]
```

Comment:
One issue with the frame schema is that `reference_frame` is written as a string and does not support objects. Reference frames (other than ICRS) need attributes to be fully defined. Is there a chance that this can support the frame_attributes dict in astropy.coordinates? I think reference_frame only needs to have additional properties allowed where the frame attributes can be dumped. I see Time already has its own definition.